src/com/ableton/Pipenv.groovy

```diff
@@ -0,0 +1,114 @@
+   *                 <strong>Required value, may not be null!</strong>
+   *               </li>
+   *               <li>
+   *                 {@code returnStatus}: Return the status code of the script.
+   *                 <strong>Note:</strong> this option and the {@code returnStdout}
+   *                 option are mutually exclusive.
+   *               </li>
+   *               <li>
+   *                 {@code returnStatus}: Return standard output from the script.
+   *                 <strong>Note:</strong> this option and the {@code returnStatus}
+   *                 option are mutually exclusive.
+   *               </li>
+   *             </ul>
+   * @return Map of return values. The keys in the map correspond to the Python versions
+   *         given in {@code args.pythonVersions}, and the values are either integers,
+   *         strings, or null, depending on whether {@code returnStatus},
+   *         {@code returnStdout}, or neither option was specified (respectively).
+   */
+  Map runCommand(Map args) {
+    assert script
+    assert args
+    assert args.command
+    assert args.pythonVersions
+
+    Map result = [:]
+
+    try {
+      args.pythonVersions.each { python ->
+        script.sh "pipenv install --dev --python ${python}"
+        result[python] = script.sh(
+          returnStatus: args.returnStatus ?: false,
+          returnStdout: args.returnStdout ?: false,
+          script: "pipenv run ${args.command}",
+        )
+      }
+    } finally {
+      try {
+        script.sh 'pipenv --rm'
+      } catch (ignored) {}
+    }
+
+    return result
+  }
+
+  /**
+   * Run a closure with Pipenv using multiple versions of Python. Because the virtualenv
+   * created by Pipenv must be wiped out between runs, this function cannot be
+   * parallelized and therefore the commands are run serially for each Python version.
+   *
+   * @param args Map of arguments. Valid arguments include:
+   *             <ul>
+   *               <li>
+   *                 {@code pythonVersions}: List of Python versions to run the command
+   *                 with. This argument is passed to Pipenv via {@code pipenv --python}.
+   *                 See {@code pipenv --help} for supported syntax.
+   *                 <strong>Required value, may not be null!</strong>
+   *               </li>
+   *             </ul>
+   * @param body Closure body to execute. The closure body is passed the python version
+   *             as an argument.
+   * @return Map of return values. The keys in the map correspond to the Python versions
+   *         given in {@code args.pythonVersions}, and the values are the result of
+   *         executing the closure body.
+   */
+  Map runWith(Map args, Closure body) {
+    assert script
+    assert args
+    assert args.pythonVersions
+
+    Map result = [:]
+
+    try {
+      args.pythonVersions.each { python ->
+        script.sh "pipenv install --dev --python ${python}"
```

Comment:
It should be documented somewhere that we are passing `--dev` to `pipenv install`.

Reply:
Fixed